### PR TITLE
Fix runtime failure when building with Intel 18.0.5.274 and Intel MPI 2018.0.4

### PR DESCRIPTION
**Pull request by rmontuoro**
## Description

When building with Intel Fortran Compiler 18.0.5.274 and Intel MPI 2018.0.4, MAPL crashes at runtime upon Cap initialization, as described in further detail in issue #861.

This PR declares as Fortran derived type (`type`) a few variables previously declared as `class` to prevent the observed runtime errors.

## How Has This Been Tested?

This PR was tested within the UFS model on NOAA R&D Hera platform.
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (affects only documentation or cleanup)

## Checklist:

- [ ] I have tested this change with a run of GEOSgcm (if non-trivial)
- [x] I have added one of the required labels (0 diff, 0 diff trivial, 0 diff structural, non 0-diff)
- [x] I have updated the CHANGELOG.md accordingly following the style of [Keep a Changelog](https://keepachangelog.com/en/1.0.0/#how)

### CODE REVIEW

**Comment by weiyuan-jiang** [comment]
The PR #846 will address this issue.  But I don't mind taking a small step here

**Comment by weiyuan-jiang** [comment]
After taking another look, I think this is OO fix we want.  MAPL_Cap here takes a MAPL_CapOptions type, but in other applications, MAPL_FlapCapoptions is passed in

File: gridcomps/Cap/MAPL_Cap.F90 [hunk 1]
```diff
@@ -29,7 +29,7 @@ module MAPL_CapMod
       integer :: rank
       integer :: npes_member
  
-      class (MAPL_CapOptions), allocatable :: cap_options
+      type (MAPL_CapOptions), allocatable :: cap_options
```
Comment (tclune) [inline gridcomps/Cap/MAPL_Cap.F90:32]: I suspect the ALLOCATABLE is no longer needed.

**Comment by mathomp4** [comment]
Per @rmontuoro This should be taken care of with MAPL 2.7.0, correct?

**Comment by rmontuoro** [comment]
MAPL 2.7.0 addressed this issue.

**Pull request closed without merging**